pipeline/authn/authenticator_cookie_session.go

```diff
@@ -204,6 +204,20 @@ func forwardRequestToSessionStore(client *http.Client, r *http.Request, cf Authe
 			return json.RawMessage{}, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Unable to fetch cookie session context from remote: %+v", err))
 		}
 		return body, nil
+	} else if res.StatusCode == http.StatusTooManyRequests {
+		// Handle rate limiting - pass through the response
+		body, _ := io.ReadAll(res.Body)
```

Comment:
I wonder if it would be possible to pass through these headers and the body? Also what about the other handlers with upstream code?